### ISSUE TITLE
notify material broken for Mercurial

### ISSUE THREAD

**Issue by FredrikWendt** [gocd/gocd#1977]
We're using SCM push hooks to notify our GoCD server of changes to source code Material. This works just fine for all git based repositories, but not hg based repositories.

Example commands used to test this bug:
``` 
[root@floating-ip-99-195 VB4rHK8WeVGZGuFH]# curl 'http://10.67.99.195:8153/go/api/material/notify/hg/' -X POST -d 'repository_url=ssh://USER@source-git.got.jeppesensystems.internal//PATH_COPIED/common/'
Unable to find material.
[root@floating-ip-99-195 VB4rHK8WeVGZGuFH]# curl 'http://10.67.99.195:8153/go/api/material/notify/hg/' -X POST -d 'repository_url=ssh://USER@source-git.got.jeppesensystems.internal//PATH_COPIED/template/service'
Unable to find material.
[root@floating-ip-99-195 VB4rHK8WeVGZGuFH]# curl 'http://10.67.99.195:8153/go/api/material/notify/git/' -X POST -d 'repository_url=ssh://git@source-git.got.jeppesensystems.internal:99922/PATH_COPIED/gocd-workshop.git'
The material is now scheduled for an update. Please check relevant pipeline(s) for status.
```

There is no copy paste issue here - the paths (as used) are copied directly from the Go config XML file (and 6 eyes watched me do this, and others have tried too).

**Comment by FredrikWendt** [comment]
This problem occurs with GoCD prior to release 16.2 and when `autoUpdate` is turned on for the material.

The message was fixed in 16.2 to be “Unable to find material. Materials must be configured not to poll for new changes before they can be used with the notification mechanism."

**Comment by FredrikWendt** [comment]
And thanks to @juhijariwala :-)
